budgetApp1/budgetApp1/UserViewModel.swift

```diff
@@ -2,20 +2,28 @@ import Foundation
 
 class UserViewModel {
     
-    // MARK: - Singleton instantiation of ViewModel
-    static let shared = UserViewModel()
+    let authenticationRepository: AuthenticationRepositoryType
     
-    private var users: [User] = []
+    private var userData = [UserEntity]()
+    let coreDataHandler = CoreDataHandler()
     
-    private init() {}
+    init(authenticationRepository: AuthenticationRepositoryType) {
+        self.authenticationRepository = authenticationRepository
+    }
+    
+    func fetchAllUsers() {
+        do {
+            try userData = coreDataHandler.context.fetch(UserEntity.fetchRequest())
+        } catch {
+            print("Error fetching users: \(AuthError.failedTofetchUsers)")
+        }
+    }
```

Comment:
after checking printing for testing try not to have print statements in your code (from experience)

Reply:
In these functions I used print statements instead of a failure result from the escaping closure because I am no longer using escaping closures for these functions as it is a bit redundant. The context does error handling already in the try-catch blocks, so having a success/failure returned would be redundant I believe. 
I hear what you are saying 100%, since I don't want to leave the catch block empty I included the print statements just for these functions though. I'll ask if I can leave the catch blocks empty for now...